### ISSUE TITLE
compile list of stuff we need money to pay for

### ISSUE THREAD

**Issue by mwookawa** [Harvard-PRINCESS/Guppy#21]
arm FVPs (software)
dev boards (hardware)
misc for dev boards

FORMAL QUOTES

**Comment by mwookawa** [comment]
armv8 and armv7 FVPs for all seats. possibly DS-5 seats

**Comment by mwookawa** [comment]
might as well throw some workstations in. iirc HUIT prefers dell and has some standard configs we can easily quote out.

i am thinking 4 workstations? these would all be shared among the team (most people will be sshing in unless they need direct dev board access).

**Comment by mwookawa** [comment]
should ask mothy if he can have a student list he hardware platforms that the nightlies are built again

**Comment by margoseltzer** [comment]
This description is incomprehensible

(I think you’re asking for something they’ve already promised; let’s not bug him)

- M

**Comment by mwookawa** [comment]
i've turned that into a full sentence, and yes, we have already asked Mothy for a list, and accordingly, we will not ask him again.

**Comment by mwookawa** [comment]
Note: Pandaboards are no longer in production and hence no longer available new. Unfortunately, this means we will not be able to obtain formal quotes on additional boards.

Currently, we have four boards, all scavenged from ebay. The average price of a pandaboardES rev. B1/2 (note that these revisions of board are required for compatibility due to a change in memory chip in revs. A and B3) is 100$. However, it is rare for a single seller to have multiple boards for sale.

Note finally that the incompatibility with the B3 revision of board was due to the EOL of all dram chips with the pinout used by the revs. B1 and B2 boards. Hence, new Chinese unbranded production of Pandaboards (average price on Ebay: 300$) may not be compatible with the barrelfish Pandaboard image that we are currently building.

Conclusion: I am personally fine with continuing to obtain used PandaboardES boards off of ebay using personal, small monies. In our formal request with formal quotes for items, I think we should consider porting barrelfish to a very similar, well-documented platform for ARMv7 and a new low-cost, well-documented platform for ARMv8 that are in production and can be vended by eg mouser.

**Comment by mwookawa** [comment]
Re: workstations, the Dell/Harvard IT standard workstation-class machine is the T7810 with dual 10-core xeons detailed here: http://www.dell.com/premier/us/en/rc1369048/#/systems/pd/rcrc1369048-4629711

Cost is 2125 per machine with 16gb of ram per workstation. It looks like additional memory can be added to the machine at manufacturer pricing without requiring us to add another quote to our request.

**Comment by mwookawa** [comment]
ARM Fixed Virtual Platform CPU simulator software is 750$ per CPU model per license. eg,

1x Cortex-A9 (Pandaboard 32-bit CPU) multiprocessor FVP license 
1x Cortex-A15 (Supported 32-bit CPU, see below for boards) multiprocessor FVP license
1x Cortex-A57 (Supported 64-bit CPU) multiprocessor FVP license

= 750$ * 3 = 2250

The Cortex-A15 is used on the Jetson TK1 (Nvidia Tegra K1), Beagleboard-X15 (TI AM5728)
The Cortex-A9 is used on the PandaboardES, the ARM UDOO, and a number of other inexpensive dev boards that use NXP's i.MX6 A9 SoC.
The Cortex-A57 is used by the Jetson TX1 (Nvidia Tegra X1) development board
The Cortex-A53 is used by numerous development boards; the cheapest such board seems to be 96boards HiKey960 (Huawei Kirin). The Cortex-A53 is the most common ARMv8 core by far.

**Comment by margoseltzer** [comment]
Have you checked if they have any academic pricing?

- M

**Comment by mwookawa** [comment]
filling out the contact sheet for a rep and quote now. i dealt with newark at draper and had a pretty agreeable rep who could work on price and was quick with proper quotes.

**Comment by mwookawa** [comment]
More prices:

Jetson TK1 (A15 32-bit) -- 129$ after EDU discount
Jetson TX1 (A57 64-bit) -- 299$ after EDU discount

both Jetson boards have a modern Nvidia GPU on the same SoC as the ARM cores -- Nvidia seems quite keen on giving away its GPUs to schools..

The hardware ARMv8 platforms mentioned by Mothy are:

> - Applied Micro Mustang X-C1 X-Gene 1 evaluation board
> - Gigabyte R150-T61 (2-socket ThunderX 1U rackmount server)
> - Gigabyte R120-T30 (1-socket ThunderX 1U rackmount server)

These are all in the thousands of dollars and use custom ARM cores with either 8 or 32-cores per chip (x-gene or ThunderX, resp)

**Comment by dh6713** [comment]
A "workstation" sits on someone's desk. A "build server" is something you ssh into. These have substantively different requirements, one being that machines used by more than one person require administration.

**Comment by dh6713** [comment]
For the record: while at this point all my machines (including my desktop) are getting oldish, I don't particularly need new ones. I am going to need a linux machine (since I'm sure barrelfish won't build hosted on netbsd given the problems with ghc) ... I could use one of the machines on the shelf but if we're going to buy some new machines anyway a new machine's not a bad idea.

My existing linux machine is a Pentium 4 and should just be thrown out, except I need to get some PASS stuff off it still. (The reason I haven't yet is that it blew up when I tried to update the OS and it can't access the network.)

also let's not buy server-grade machines intending to put them in MD121 without taking some care about noise levels...

**Comment by mwookawa** [comment]
7810s are about 5u thick and most of the box is heatsink. They are very
quiet. The only major downside to them is that they only have 4 dimms per
socket and they are full of proprietary dell junk.

Regardless, with current phase change (sintered heat pipe and vapor
chamber) based heatsinks, you only need about 150-160mm of radiator space
above the cpu socket to dissipate 200-250W at office noise levels. all
workstation cases basically have to be this thick to fit modern GPUs.

in short, modern workstation-class machines can run more compute than
racked servers while still being office-noise friendly.

**Comment by mwookawa** [comment]
OK, so I've RFQ'd ARM on the FVP suite via various public webforms a few times now with no responses. i just shot off a much more detailed email requesting academic pricing to what seems to be a lead-generating email address and cc'd susan as well. however, given that we are pressed for time with various people going on vacation, we should probably just put the public prices into our request and loop back if there is an issue. There is a public SKU which is the smallest package that has strictly more than we need:

http://www.newark.com/arm/ds5ue-kd-40001/s-w-compiler-arm-cortex-a-m-r/dp/45X8229

6600. one year license, floating so that everyone on the team can use the tools. This includes a simulated virtual platform for every current ARM IP core as well as the closed source (LLVM-based) ARM compiler suite with heavily customized assembler, disassembler, frighteningly robust debugger, and support for most hardware debug devices, including the frighteningly good one that ARM vends for a couple thousand more.

**Comment by mwookawa** [comment]
have finally made a tiny bit of progress on getting academic program contacts and pricing for some of these things.

**Comment by margoseltzer** [comment]
Thanks!  Sorry I have not been able to sync up on this …

Tomorrow I hope

- M

**Comment by mwookawa** [comment]
i'm moving this discussion to the documents repo under Documents#1
